src/main/scala/firrtl/passes/CheckCombLoops.scala

```diff
@@ -104,29 +104,42 @@ object CheckCombLoops extends Pass {
   }
 
   /*
-   * Recover the full path of the cycle. Since cycles may pass through
-   * simplified instances, the hierarchy that the path passes through
-   * must be recursively recovered.
+   * Recover the full path from a path passing through simplified
+   * instances. Since edges may pass through simplified instances, the
+   * hierarchy that the path passes through must be recursively
+   * recovered.
    */
-  private def recoverCycle(
+  private def expandInstancePaths(
     m: String,
     moduleGraphs: mutable.Map[String,DiGraph[LogicNode]],
     moduleDeps: Map[String, Map[String,String]], 
     prefix: Seq[String],
-    cycle: Seq[LogicNode]): Seq[String] = {
+    path: Seq[LogicNode]): Seq[String] = {
     def absNodeName(prefix: Seq[String], n: LogicNode) =
       (prefix ++ n.inst ++ n.memport :+ n.name).mkString(".")
-    val cycNodes = (cycle zip cycle.tail) map { case (a, b) =>
+    val pathNodes = (path zip path.tail) map { case (a, b) =>
       if (a.inst.isDefined && !a.memport.isDefined && a.inst == b.inst) {
         val child = moduleDeps(m)(a.inst.get)
         val newprefix = prefix :+ a.inst.get
         val subpath = moduleGraphs(child).path(b.copy(inst=None),a.copy(inst=None)).tail.reverse
-        recoverCycle(child,moduleGraphs,moduleDeps,newprefix,subpath)
+        expandInstancePaths(child,moduleGraphs,moduleDeps,newprefix,subpath)
       } else {
         Seq(absNodeName(prefix,a))
       }
     }
-    cycNodes.flatten :+ absNodeName(prefix, cycle.last)
+    pathNodes.flatten :+ absNodeName(prefix, path.last)
+  }
+
+  private def findCycleInSCC[T](sccGraph: DiGraph[T]): Seq[T] = {
```

Comment:
A one-liner describing what this does would be great, also mentioning that you've already pruned the graph, I was confused by the fact that you only visit the head edge of a given vertex.